File: 10347/medians.c
#include <stdio.h>
#include <math.h>

// Accepted

int main(int argc, char *argv[])
{
	double a, b, c;
	double s, area;
	double t;

	while(scanf("%lf %lf %lf\n", &a, &b, &c) == 3)
	{
		s = (a + b + c) / 2;
		t = s * (s - a) * (s - b) * (s - c);

		if(t <= 0)
			area = -1;
		else
			area = (4 * sqrt(t)) / 3;

		printf("%.3lf\n", area);
	}

	return 0;
}

File: 10347/medians.py
import sys
import math

for line in sys.stdin:
	x = line.strip().split()

	a = float(x[0])
	b = float(x[1])
	c = float(x[2])

	s = (a + b + c) / 2

	area = (math.sqrt(s * (s - a) * (s - b) * (s - c)) * 4) / 3

	print("%.3f"%(area))
